docs/beman-standard.md

```diff
@@ -0,0 +1,170 @@
+2. **[CORE.PRODUCTION_READY] Production-ready**. Production feedback
+   necessitates reliable, well-documented software.
+3. **[CORE.INDUSTRY_STANDARD] Industry standard technology**. Where there's
+   industry consensus on best practices, we should take advantage. Innovation in
+   tooling and style is not our purpose.
+4. **[CORE.INCLUSIVE] Welcoming and inclusive community**. Broad, useful
+   feedback requires an unobstructed path for using, reviewing, and
+   contributing to Beman libraries. This principle encompasses ergonomics,
+   cross-industry participation, and cultural accommodation.
+
+### Changing this document
+
+This is a living document that adapts to evolving best practices and community
+needs. To make changes:
+
+1. Create a [discourse topic](https://discourse.boost.org) detailing the change
+   and how it aligns with the core principles.
+2. After some community discussion, create a PR with the actual change on
+   [GitHub](https://github.com/beman-project/beman) with a *leads question*
+   label. The PR should also link to the discourse topic.
+3. Continue discussions on the PR and discourse topic.
+4. Await a leads a decision based on the community feedback.
+
+### Conventions
+
+This standard consists of entries that include an all-caps, dot-separated
+identifier for referencing.
+
+With the exception of the core principles, these entries are either rules or
+recommendations.
+
+* **Rules** must be followed in order to conform to this standard. These entries
+  are prefixed by "RULE:".
+* **Recommendations** should be followed in general, but specific circumstances
+  may make this a less-than-ideal choice. Libraries not following a specific
+  recommendation can still conform to this standard. These entries are prefixed
+  by "RECOMMENDATION:".
+
+## License
+
+**[LICENSE.APPROVED]** REQUIREMENT: All Beman libraries must be licensed
+under an approved license. These are:
+
+1. [Apache License v2.0 with LLVM Exceptions](https://llvm.org/LICENSE.txt)
+2. [Boost Software License 1.0](https://www.boost.org/LICENSE_1_0.txt)
+3. [The MIT License](https://opensource.org/license/mit)
+
+**[LICENSE.BOOST]** RECOMMENDATION: All Beman libraries should be licensed under
+the [Apache License v2.0 with LLVM Exceptions](https://llvm.org/LICENSE.txt).
+
+**[LICENSE.CRITERIA]** REQUIREMENT: All approved licenses must meet the
+following requirements:
+
+1. Simple to read and understand.
+2. Permission without fee to copy, use and modify the software for any
+   use (commercial and non-commercial).
+3. Requires that the license appears on all copies of the software source code.
+4. Must not require that the license appears with executables or other binary
+   uses of the library.
+5. Must not require that the source code be available for execution or other
+   binary uses of the library.
+
+## General
+
+**[LIBRARY_NAMES]** REQUIREMENT: Beman libraries names begin with `Beman.`
```

Comment:
I think this has to be a recommendation instead of a requirement -- or we have to have some way to reference things from Beman without impacting the core library.  For example the scnlib that we're currently including already has a name that doesn't follow the convention.

Reply:
@eliaskosunen would you object to calling your library Beman.Scnlib or something like that if it gets incorporated into the Beman project.

Reply:
It's a good practice to have names all correspond across the entire library. In the sense of physical design and logical design "matching".

Including: the repo name, project name, header include targets, library target name in CMake, library file name, C++ namespace name, macro name prefix if applicable, extern C names, and maybe some other things.

Do we want `beman::foobar::some_range` and `find_package(beman-foobar)` and `#include <beman/foobar/core.hpp>` and `libbeman_foobar.a` and so on?

What if I library ends up gaining its own life beyond the standard? Some libraries could reasonably be declined for standardization and "only" end up being very popular libraries like nlohmanm-json, googletest, or cli11. Do we expect a "hard fork" in those cases, renaming everything to de-bemanize it?

I don't see a technical advantage to the Beman to level scope compared to just using a unique enough library name. I would expect good docs and developer support and a helpful community would speak for itself if branding is the concern. But I guess I'm an engineer and not a marketing guy.

Reply:
I changed this requirement to a recommendation. Some libraries have existing branding or have use cases outside the context of the Beman project. Such libraries should be permitted in the Beman project.

However, to address @bretbrownjr 's concern, we are keeping the requirement that Beman libraries have a unified interface (e.g. `#include <Beman/ShortName/foo.hpp`) so Beman users know exactly what is being proposed for the standard. Additionally, after this initial PR, I think we should have others that recommend unified CMake package names, library names, etc.

Reply:
You might consider requiring dual-licensing.  IIUC the Boost License is less constraining than the other one, with many fewer [redistribution requirements (section 4)](https://apache.org/licenses/LICENSE-2.0).   FWIW, Beman really cared about lowering the bar for usage.